scripts/e2e-host-controller-runtime.sh

```
@@ -0,0 +1,329 @@
+  [[ -d "$source_dir" && "$source_dir" == /* ]] || fail "scenario source directory must be an absolute directory"
+  local tested_commit
+  tested_commit="$(git -C "$source_dir" rev-parse HEAD 2>/dev/null || true)"
+  validate_revision "$tested_commit"
+  validate_registered_scenario "$source_dir" "$scenario" "$phase"
+
+  local project
+  project="$(managed_project_name "$run_id" "$attempt" "$phase" "$stack_scenario")"
+  local helpers
+  helpers="$(scenario_helpers "$source_dir" "$phase" "$scenario")" || fail "scenario helper profiles are unavailable"
+  [[ "$helpers" =~ ^[a-z0-9_,]*$ ]] || fail "invalid helper profile list"
+  local created_at="$(date -u +%Y-%m-%dT%H:%M:%SZ)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)